terraform/ehr_deletion_lambda.tf

```diff
@@ -0,0 +1,96 @@
+resource "aws_lambda_function" "ehr_hard_deletion" {
+  filename      = var.ehr_hard_deletion_lambda_zip
+  function_name = "${var.environment}-ehr-hard-deletion-lambda"
+  role          = aws_iam_role.ehr_hard_deletion_lambda_role.arn
+  handler       = "main.lambda_handler"
+  source_code_hash = filebase64sha256(var.ehr_hard_deletion_lambda_zip)
+  runtime = "python3.12"
+  timeout = 15
+  tags = {
+    Environment = var.environment
+    CreatedBy   = var.repo_name
+    Terraform = "True"
+  }
+  environment {
+    variables = {
+      S3_REPO_BUCKET=data.aws_s3_bucket.ehr_repo_bucket.bucket
+#      S3_LARGE_MESSAGES_BUCKET=data.aws_s3_bucket.xxx.bucket
+    }
+  }
+}
+
+resource "aws_lambda_event_source_mapping" "hard_delete" {
```

Comment:
This name should include the DynamoDB table name, too e.g. `ehr_transfer_tracker_dynamodb_to_hard_delete_lambda`